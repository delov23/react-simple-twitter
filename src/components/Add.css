form.centered-form {
    width: 65vw;
    height: 40vh;
    margin-top: 20px !important;
    border: 0.1vw solid #35414a;
    border-top: 0.1vw solid #35414a !important;
    border-radius: 10px;
    -webkit-border-radius: 10px;
    -moz-border-radius: 10px;
    -ms-border-radius: 10px;
    -o-border-radius: 10px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color: #0f171f;
}

form.centered-form input.form-input {
    margin-bottom: 2vh;
    width: 80%;
    margin-left: auto;
    margin-right: auto;
    height: 2vh;
    font-family: 'Raleway', sans-serif;
    color: black;
    font-size: 1.1rem;
    background-color: #ffffff;
    border: 0.1vw solid #242b31;
    border-radius: 12px;
    padding: 1vh;
    -webkit-border-radius: 12px;
    -moz-border-radius: 12px;
    -ms-border-radius: 12px;
    -o-border-radius: 12px;
}

div.app-wrapper.dark form.centered-form input.form-input {
    background-color: #273c50;
    color: #dfdfdf;
}

form.centered-form textarea.form-input {
    margin-bottom: 2vh;
    width: 80%;
    max-width: 80%;
    margin-left: auto;
    margin-right: auto;
    font-family: 'Raleway', sans-serif;
    color: black;
    font-size: 1.1rem;
    background-color: #ffffff;
    border: 0.1vw solid #242b31;
    border-radius: 12px;
    padding: 1vh;
    -webkit-border-radius: 12px;
    -moz-border-radius: 12px;
    -ms-border-radius: 12px;
    -o-border-radius: 12px;
}

div.app-wrapper.dark form.centered-form textarea.form-input {
    background-color: #273c50;
    color: #dfdfdf;
}

input.lg {
    height: 7vh !important;
}

form.centered-form button.form-submit {
    cursor: pointer;
    font-family: 'Raleway', sans-serif;
    color: black;
    font-size: 1.1rem;
    background-color: white;
    border: 0.1vw solid #242b31;
    border-radius: 12px;
    width: 40%;
    margin-left: auto;
    margin-right: auto;
    padding: 1vh;
    -webkit-border-radius: 12px;
    -moz-border-radius: 12px;
    -ms-border-radius: 12px;
    -o-border-radius: 12px;
}

div.app-wrapper.dark form.centered-form button.form-submit {
    background-color: #273c50;
    color: #dfdfdf;
}

@media screen and (min-width: 1300px) {
    form.centered-form {
        width: 40vw;
    }
}
