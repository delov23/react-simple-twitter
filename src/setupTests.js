import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { cleanup } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import 'jest-enzyme';

configure({ adapter: new Adapter() });

const mockStore = configureStore([]);

const store = mockStore({
    tweets: [],
    theme: 'dark'
});

export const Wrapper = () => <Provider store={store} />;

afterEach(() => {
    cleanup();
});
