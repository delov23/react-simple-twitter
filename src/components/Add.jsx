import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';

import { connect } from 'react-redux';
import { addTweet } from '../state/actions';
import { getUuid } from '../util/uuid';

import './Add.css';

class Add extends Component {
    state = {
        tweet: {
            id: getUuid(),
            user: {
                name: 'Tom Newman',
                username: 'tom_nm2',
                picture:
                    'https://www.pixelite.co.nz/content/images/2019/09/mateo-avila-chinchilla-x_8oJhYU31k-unsplash.jpg'
            },
            content: '',
            image: '',
            timestamp: Date.now()
        },
        redirect: false
    };

    handleChange = ev => {
        this.setState({
            tweet: { ...this.state.tweet, [ev.target.name]: ev.target.value }
        });
    };

    addTweet = () => {
        this.setState({
            tweet: { ...this.state.tweet, timestamp: Date.now() },
            redirect: true
        });
        this.props.addTweetToState(this.state.tweet);
    };

    render() {
        if (this.state.redirect) {
            return <Redirect to={'/'} />;
        }

        return (
            <form
                className="centered-form content-block"
                onSubmit={this.addTweet}
            >
                <h3>Say something to your followers!</h3>
                <textarea
                    rows="4"
                    minLength="10"
                    maxLength="120"
                    required
                    className="form-input lg"
                    type="text"
                    placeholder="Content"
                    name="content"
                    onChange={this.handleChange}
                    value={this.state.tweet.content}
                />
                <input
                    className="form-input"
                    type="text"
                    placeholder="Image"
                    name="image"
                    onChange={this.handleChange}
                    value={this.state.tweet.image}
                />
                <button type="submit" className="form-submit">
                    Add tweet
                </button>
            </form>
        );
    }
}

export default connect(
    null,
    { addTweetToState: addTweet }
)(Add);

/* 
const Add = () => {
    const ctx = useContext(TweetsContext);
    const [redirect, setRedirect] = useState(false);
    const [tweet, setTweet] = useState({
        id: getUuid(),
        user: {
            name: 'Tom Newman',
            username: 'tom_nm2',
            picture:
                'https://www.pixelite.co.nz/content/images/2019/09/mateo-avila-chinchilla-x_8oJhYU31k-unsplash.jpg'
        },
        content: '',
        image: '',
        timestamp: Date.now()
    });

    const addTweet = () => {
        tweet.timestamp = Date.now();
        ctx.addTweet(tweet);
        setRedirect(true);
    };

    const handleChange = ev => {
        setTweet({ ...tweet, [ev.target.name]: ev.target.value });
    };

    if (redirect) {
        return <Redirect to={'/'} />;
    }

    return (
        <form className="centered-form content-block" onSubmit={addTweet}>
            <h3>Say something to your followers!</h3>
            <textarea
                rows="4"
                minLength="10"
                maxLength="120"
                required
                className="form-input lg"
                type="text"
                placeholder="Content"
                name="content"
                onChange={handleChange}
                value={tweet.content}
            />
            <input
                className="form-input"
                type="text"
                placeholder="Image"
                name="image"
                onChange={handleChange}
                value={tweet.image}
            />
            <button type="submit" className="form-submit">Add tweet</button>
        </form>
    );
};

export default Add;

*/
