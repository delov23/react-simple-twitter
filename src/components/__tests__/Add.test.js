import React from 'react';
import { shallow } from 'enzyme';
import { Wrapper } from '../../setupTests';

import Add from '../Add';


describe('Add component', () => {
    it('should render', () => {
        const component = shallow(<Wrapper><Add /></Wrapper>);

        expect(component).toMatchSnapshot();
    });
});
