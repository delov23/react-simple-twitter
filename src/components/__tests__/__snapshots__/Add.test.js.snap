// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add component should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
